nupic/swarming/hypersearch/HyperSearchState.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2015, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+
+
+class HyperSearchState(object):
+  """
+    Defines the class storing necessary state for hyperSearch runs.
+  """
+
+  def __init__(self, permuteVars, particles, numParticles, options):
+    """
+      @oaram  permuteVars   (list)    Collection of client-specified permutation
+                                      variables necessary for a given run.
+      @param  particles     (list)    Collection of particles instantiated by
+                                      client, in HyperSearchRunner.
+      @param  numParticles  (int)     Client provided number of particles in
+                                      swarm. TODO: May want to just provide a
+                                      generic param with swarm-specific options?
+      @param  options       (dict)    Dict of client-specified params for
```

Comment:
This should either be broken up into separate kwargs, or be an instance of a specific class representing possible parameters.